src/util/psbt/serialize.rs

```diff
@@ -363,6 +363,77 @@ impl Deserialize for TapTree {
     }
 }
 
+impl Serialize for PartiallySignedTransaction {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf: Vec<u8> = Vec::new();
+
+        buf.extend(b"psbt");
+
+        buf.push(0xff_u8);
+
+        self.consensus_encode_map(&mut buf).unwrap();
+
+        for i in &self.inputs {
+            i.consensus_encode(&mut buf).unwrap();
+        }
+
+        for i in &self.outputs {
+            i.consensus_encode(&mut buf).unwrap();
+        }
+
+        buf
+    }
+}
+
+// impl Decodable for PartiallySignedTransaction {
+impl Deserialize for PartiallySignedTransaction {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        let magic = &bytes[0..4];
+
+        if *b"psbt" != magic {
+            return Err(Error::InvalidMagic.into());
+        }
+
+        if 0xff_u8 != bytes[4] {
+            return Err(Error::InvalidSeparator.into());
+        }
+
+
+        let mut d = &bytes[4..];
```

Comment:
There is an off-by-one error here, should be a 5 - 4 bytes of magic, 1 byte of 0xff